packages/mdc-auto-init/index.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// tslint:disable:only-arrow-functions
+
+import {MDCComponent} from '@material/base/component';
+import {MDCFoundation} from '@material/base/foundation';
+
+interface ComponentClass {
```

Comment:
I went back and forth on this, but now that you point it out, `register()` has an argument with this type, so we need to export it.

Renamed the interface to `MDCComponentClass`. People might confuse it with `MDCComponent`, though—is there a better name for this?

Reply:
What about `AutoInit`? Components need to implement this interface in order for attachTo to work.

Reply:
What about MDCAttachable if we're going for interface-y names? Something ending in -Class definitely seems confusing for that purpose.

Reply:
Also RE Matt's comment about needing to implement this interface... I wondered about that, but it seems like our components Just Work by inference without having MDCComponent explicitly implement this interface?

Reply:
I like `MDCAttachable`. Done!

True, our components Just Work by inference, but do we think clients will need to declare variable of the interface type? (Similar to what we do for adapters in [`getDefaultFoundation()`](https://github.com/material-components/material-components-web/blob/d349fea873f466981846b0f6c9714f37bede78e3/packages/mdc-checkbox/component.ts#L102-L118))

Reply:
This doesn't look like it was actually done on the branch, so I just pushed it.

I agree we should export it since it's referenced in a public API, but I don't expect anyone to need to explicitly reference it if even we don't need to.